syntax = "proto3";

package apexlegends;

message Vec3 {
  float x = 1;
  float y = 2;
  float z = 3;
}

message Badge {
  int32 badge_index = 1;
  int32 badge_data = 2;
}

message PlayerState {
  Vec3 origin = 2;
  Vec3 view_angles = 3;
  Vec3 velocity = 4;
  int32 health = 5;
  int32 shield = 6;
  int32 max_health = 7;
  int32 max_shield = 8;
  int32 helmet_type = 9;
  int32 armor_type = 10;
  int32 team_num = 11;
  int32 xp = 12;
  int32 flags = 13;
  bool is_alive = 14;
  bool is_knocked = 15;
  int32 love_state = 16;
  int32 active_weapon = 17;
  string player_name = 18;
  uint64 platform_uid = 19;
  uint64 eadp_uid = 20;
  Vec3 head_position = 21;
  bool controller_active = 22;
  int32 character_index = 23;
  repeated Badge badges = 24;
  int32 kills = 25;
  int32 damageDealt = 26;
  bool kill_leader = 27;
  bool winning_team = 28;
  float yew = 29;
  int32 team_member_index = 30;
}

message Players {
  int32 version = 1;
  repeated PlayerState players = 2;
}
